From a3261fdfd163a4e094c3b5b778928842e6a2e997 Mon Sep 17 00:00:00 2001
From: Jurijs Soloveckis <jsoloveckis@maxlinear.com>
Date: Wed, 8 Nov 2023 10:34:59 +0200
Subject: [PATCH] WLANRTSYS-60035: Puma8: Align nl80211.h to latest from
 opensource

Sync with wireless-next.git include/uapi/linux/nl80211.h
This brings in nl80211 definitions as of 2023-10-23.
---
 src/drivers/mxl_driver_nl80211.h |  3 ++-
 src/drivers/mxl_nl80211.h        | 20 ++++++++++++++++++++
 src/drivers/nl80211_copy.h       | 16 +++++++++++++---
 3 files changed, 35 insertions(+), 4 deletions(-)

diff --git a/src/drivers/mxl_driver_nl80211.h b/src/drivers/mxl_driver_nl80211.h
index d64e1e47d..ffa591902 100644
--- a/src/drivers/mxl_driver_nl80211.h
+++ b/src/drivers/mxl_driver_nl80211.h
@@ -18,7 +18,8 @@
 #include "drivers/nl80211_copy.h"
 
 #define MXL_NL80211_CMD_C2S \
-	C2S(NL80211_CMD_SET_DFS_STATE)
+	C2S(NL80211_CMD_SET_DFS_STATE) \
+	C2S(NL80211_CMD_ML_CHAN_SWITCH_DONE_NOTIFY)
 
 int mxl_nl80211_set_channel_dfs_state(void *priv, enum nl80211_dfs_state dfs_state,
 				      int freq, int dfs_time, int dfs_debug);
diff --git a/src/drivers/mxl_nl80211.h b/src/drivers/mxl_nl80211.h
index 56dc68748..b9ae0c502 100644
--- a/src/drivers/mxl_nl80211.h
+++ b/src/drivers/mxl_nl80211.h
@@ -20,6 +20,7 @@
  */
 #define MXL_NL80211_CMDS			\
 	NL80211_CMD_SET_DFS_STATE,		\
+	NL80211_CMD_ML_CHAN_SWITCH_DONE_NOTIFY,		\
 
 
 /*
@@ -62,7 +63,26 @@
 	NL80211_ATTR_ML_STA_INFO,		\
 	NL80211_ATTR_HEOPER_COLOR_OFFS_BEACON,	\
 	NL80211_ATTR_COLOR_CHANGE_SWITCH_TIME,	\
+	NL80211_ATTR_ML_CHAN_SWITCH_DONE_TYPE,	\
 
+/*
+ * @NL80211_STA_INFO_MAX_RSSI: @max_rssi filled
+ */
+#define MXL_NL80211_STA_INFO		\
+	NL80211_STA_INFO_MAX_RSSI,		\
+
+/*
+ * @@NL80211_BAND_ATTR_MULTILINK_SUPPORT: Multilink capabilites
+ */
+#define MXL_NL80211_BAND_ATTR		\
+	NL80211_BAND_ATTR_MULTILINK_CAPA,		\
+
+/*
+ * @NL80211_BSS_NOISE: noise level when receiving probe response/beacon
+ *	frames in dBm (s8).
+ */
+#define MXL_NL80211_BSS		\
+	NL80211_BSS_NOISE,		\
 
 /*
  * @NL80211_RADAR_CAC_PAUSED: Channel Availability Check has been paused.
diff --git a/src/drivers/nl80211_copy.h b/src/drivers/nl80211_copy.h
index 2cb54389c..67b06c30a 100644
--- a/src/drivers/nl80211_copy.h
+++ b/src/drivers/nl80211_copy.h
@@ -1584,9 +1584,6 @@ enum nl80211_commands {
 	NL80211_CMD_SET_FILS_AAD,
 
 	NL80211_CMD_ASSOC_COMEBACK,
-#ifdef CONFIG_VENDOR_MXL
-	MXL_NL80211_CMDS
-#endif /* CONFIG_VENDOR_MXL */
 
 	NL80211_CMD_ADD_LINK,
 	NL80211_CMD_REMOVE_LINK,
@@ -1604,6 +1601,9 @@ enum nl80211_commands {
 	NL80211_CMD_ASSOC_MLO_RECONF,
 	NL80211_CMD_EPCS_CFG,
 
+#ifdef CONFIG_VENDOR_MXL
+	MXL_NL80211_CMDS
+#endif /* CONFIG_VENDOR_MXL */
 	/* add new commands above here */
 
 	/* used to define NL80211_CMD_MAX below */
@@ -3992,6 +3992,9 @@ enum nl80211_sta_info {
 	NL80211_STA_INFO_AIRTIME_LINK_METRIC,
 	NL80211_STA_INFO_ASSOC_AT_BOOTTIME,
 	NL80211_STA_INFO_CONNECTED_TO_AS,
+#ifdef CONFIG_VENDOR_MXL
+	MXL_NL80211_STA_INFO
+#endif /* CONFIG_VENDOR_MXL */
 
 	/* keep last */
 	__NL80211_STA_INFO_AFTER_LAST,
@@ -4223,6 +4226,10 @@ enum nl80211_band_attr {
 	NL80211_BAND_ATTR_S1G_MCS_NSS_SET,
 	NL80211_BAND_ATTR_S1G_CAPA,
 
+#ifdef CONFIG_VENDOR_MXL
+	MXL_NL80211_BAND_ATTR
+#endif /* CONFIG_VENDOR_MXL */
+
 	/* keep last */
 	__NL80211_BAND_ATTR_AFTER_LAST,
 	NL80211_BAND_ATTR_MAX = __NL80211_BAND_ATTR_AFTER_LAST - 1
@@ -5277,6 +5284,9 @@ enum nl80211_bss {
 	NL80211_BSS_MLD_ADDR,
 	NL80211_BSS_USE_FOR,
 	NL80211_BSS_CANNOT_USE_REASONS,
+#ifdef CONFIG_VENDOR_MXL
+	MXL_NL80211_BSS
+#endif /* CONFIG_VENDOR_MXL */
 
 	/* keep last */
 	__NL80211_BSS_AFTER_LAST,
-- 
2.43.0

